### ISSUE TITLE
False-positive unnecessary-pass on function with docstring

### ISSUE THREAD

**Issue by EugeneZelenko** [pylint-dev/pylint#8874]
### Bug description

I got false-positive for `unnecessary-pass` when function contains only docstring:

```
def function():
    '''Some docstring.'''
    pass
```

### Command used

```shell
pylint --disable=all --enable=unnecessary-pass pylint-test.py
```

### Pylint output

```shell
pylint-test.py:3:4: W0107: Unnecessary pass statement (unnecessary-pass)
```

### Expected behavior

No message.

### Pylint version

```shell
pylint 2.17.4
astroid 2.15.5
Python 3.11.4 (main, Jun 26 2023, 16:57:25) [GCC 7.5.0]
```

**Comment by EugeneZelenko** [comment]
I think, docstring is not actual code, so `pass` is still better expression that function does nothing.

**Comment by mbyrnepr2** [comment]
One thing we could improve is the [documentation](https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/unnecessary-pass.html) for this checker. It uses the abstract `Foo` convention currently.
That could be updated to be a more real-world example such as inheriting from the `Exception` class as in [this piece of Pylint source code](https://github.com/pylint-dev/pylint/blob/main/pylint/exceptions.py#L8).

**Comment by Pierre-Sassoulas** [comment]
Do you want to do the little change ? I think an exception class makes a lot of sense.
